### ISSUE TITLE
[BUG] Failure to move items to trash -- [Errno 18] Invalid cross-device link

### ISSUE THREAD

**Issue by lemstress** [paperless-ngx/paperless-ngx#615]
**Describe the bug**
While running within docker via the `linuxserver/paperless-ngx` image on a Synology NAS host system, `PAPERLESS_TRASH_DIR` and therefore the new "move to trash" feature that was introduced does not work and the files are still permanently deleted instead of moving to a specified trash directory.

**To Reproduce**
1. Within the settings for the `linuxserver/paperless-ngx` image, mount a volume from a Synology recycle folder to a folder called /recycle within the container. In my case I'm mounting the host folder `/Network Scans/#recycle` to the container folder `/recycle`.
2. Set the `PAPERLESS_TRASH_DIR:` environmental value to the recycle folder name that you specified in the previous step, in this case, `/recycle`.
3. Start the container
4. Attempt to delete a file within the Paperless interface. It deletes the file completely instead of moving the file to the specified trash directory and spits out an error in the logs. See error below:

**Expected behavior**
Paperless moves the file to a predetermined "recycle" folder as intended and does not spit an error within the log file.

**Webserver logs**
```
[ERROR] [paperless.handlers] Failed to move /media/documents/originals/0000726.pdf to trash at /recycle/0000726.pdf: [Errno 18] Invalid cross-device link: '/media/documents/originals/0000726.pdf' -> '/recycle/0000726.pdf'. Skipping cleanup!
```

**Relevant information**
 - Host OS: `DSM 7.0.1-42218 Update 3`
 - Docker Image: `linuxserver/paperless-ngx`
 - Browser: `Brave v1.37.109 - Chromium: 100.0.4896.60 (Official Build) (64-bit)`
 - Paperless-ngx verison: `1.6.0`
 - Installation method: Docker

**Note:** From the brief Google searching I did, [this stackoverflow page](https://stackoverflow.com/questions/42392600/oserror-errno-18-invalid-cross-device-link) references the error and talks about using the call `shutil.move` to move files instead of `os.rename`, as `os.rename` apparently only works when moving files within the same file system. 

I'm assuming maybe this is happening because the file system within the docker image and the native file system from the host (DSM 7 in this case) are different. I'm not sure if that applies to how paperless-ngx operates, but I thought I'd include it. Also I'm not sure if this is related to the docker image itself, or paperless-ngx.

**Comment by shamoon** [comment]
Related to #24 
@tribut interested in looking at this?

**Comment by stumpylog** [comment]
Quick search suggests using `shutil.move` instead of a rename to support cross file system

**Comment by tribut** [comment]
Thanks for the detailed report. The fix in #617 is correct.

@lemstress When "delete to trash" fails like this, it skips the deleting of the originals. So your files should still be present in the originals folder, just not reachable from the web anymore.

**Comment by lemstress** [comment]
@tribut Oh I had no idea! Thanks for the fix! Can't wait for the next release!
